ui/src/components/Publication/Creation/MainText.tsx

```diff
@@ -161,49 +159,49 @@ const MainText: React.FC = (): React.ReactElement | null => {
                 continue;
             }
 
-            // Pattern using regex groups to retrieve reference text and to determine whether a string contains DOI or URI
-            const pattern =
-                /(?<TEXT>.+?(?=http))((?<DOI>((((http|https):\/\/)(([-a-zA-Z0-9_]{1,265}([^\s]+)))))(10{1}\.([^\n]+)))|(?<URL>((((http|https):\/\/)(([-a-zA-Z0-9_]{1,265}([^\n]+)))))))|(?<TEXTONLY>.+?(?=$))/g;
-
-            const matches = textContent.matchAll(pattern);
-
-            // Iterate through matches
-            if (matches) {
-                for (const match of matches) {
-                    let location: string | null;
-                    let type: Interfaces.ReferenceType;
-                    let text = currentParagraph; // original html string
-
-                    if (match?.groups?.DOI) {
-                        type = 'DOI';
-                        location = match.groups.DOI;
-                        text = text.replace(
-                            match.groups.DOI,
-                            `<a href="${match.groups.DOI}" target="_blank" rel="noreferrer noopener">${match.groups.DOI}</a>`
-                        );
-                    } else if (match?.groups?.URL) {
-                        type = 'URL';
-                        location = match.groups.URL;
-                        text = text.replace(
-                            match.groups.URL,
-                            `<a href="${match.groups.URL}" target="_blank" rel="noreferrer noopener">${match.groups.URL}</a>`
-                        );
-                    } else {
-                        type = 'TEXT';
-                        location = null;
-                    }
-
-                    const newReference: Interfaces.Reference = {
-                        id: cuid(),
-                        publicationId,
-                        type,
-                        text,
-                        location
-                    };
+            const URL_REGEX = /(http|ftp|https):\/\/([\w_-]+(?:(?:\.[\w_-]+)+))([\w.,@?^=%&:\/~+#-]*[\w@?^=%&\/~+#-])/g;
```

Comment:
updating to (http|ftp|https):\/\/([\w_-]+(?:(?:\.[\w_-]+)+))([\w.,()<>;@?^=%&:\/~+#-]*[\w@?^=%&\/~+#-])

matches e.g.

https://doi.org/10.1002/(SICI)1096-9136(199801)15:1<53::AID-DIA521>3.0.CO;2-V
and also ignores trailing ) on 

https://doi.org/10.1002/(SICI)1096-9136(199801)15:1<53::AID-DIA521>3.0.CO;2-V)

but not

https://doi.org/10.1002/(SICI)1096-9136(199801)